analyzers/src/SonarAnalyzer.CSharp/Rules/MultilineBlocksWithoutBrace.cs

```diff
@@ -28,143 +28,113 @@
+            && !IsStatementCandidateLoop(lastStatementInIfChain))
         {
-            if (ifStatement.GetPrecedingIfsInConditionChain().Any())
-            {
-                return;
-            }
-
-            if (IsNestedStatement(ifStatement.Statement))
-            {
-                return;
-            }
-
-            var lastStatementInIfChain = GetLastStatementInIfChain(ifStatement);
-            if (IsStatementCandidateLoop(lastStatementInIfChain))
-            {
-                return;
-            }
-
             CheckStatement(context, lastStatementInIfChain, "conditionally", "unconditionally");
         }
+    }
 
-        private static bool IsNestedStatement(StatementSyntax nested) =>
-            nested.IsKind(SyntaxKind.IfStatement) ||
-            nested.IsKind(SyntaxKind.ForStatement) ||
-            nested.IsKind(SyntaxKind.ForEachStatement) ||
-            nested.IsKind(SyntaxKind.WhileStatement);
+    private static StatementSyntax LastStatementInIfChain(IfStatementSyntax ifStatement)
+    {
+        var statement = ifStatement.Statement;
 
-        private static StatementSyntax GetLastStatementInIfChain(IfStatementSyntax ifStatement)
+        while (ifStatement is { })
         {
-            var currentIfStatement = ifStatement;
-            var statement = currentIfStatement.Statement;
-            while (currentIfStatement != null)
+            if (ifStatement.Else is null)
             {
-                if (currentIfStatement.Else == null)
-                {
-                    return currentIfStatement.Statement;
-                }
-
-                statement = currentIfStatement.Else.Statement;
-                currentIfStatement = statement as IfStatementSyntax;
+                return ifStatement.Statement;
             }
-
-            return statement;
-            //var currentIfStatement = ifStatement;
-            //var statement = currentIfStatement.Statement;
-            //while (currentIfStatement != null)
-            //{
-            //    if (currentIfStatement.Else == null)
-            //    {
-            //        return currentIfStatement.Statement;
-            //    }
-
-            //    statement = currentIfStatement.Else.Statement;
-            //    currentIfStatement = statement as IfStatementSyntax;
-            //}
-
-            //return statement;
+            statement = ifStatement.Else.Statement;
+            ifStatement = statement as IfStatementSyntax;
         }
+        return statement;
+    }
 
-        private static void CheckStatement(SyntaxNodeAnalysisContext context, StatementSyntax statement,
-            string executed, string execute)
+    private static void CheckStatement(SyntaxNodeAnalysisContext context, StatementSyntax first, string executed, string execute)
+    {
+        if (SecondStatement(context.Node, first) is { } second
+            && NotEmpty(first) && NotEmpty(second))
```

Comment:
Should be single line (or 3). It's short, let's go a single
```suggestion
        if (SecondStatement(context.Node, first) is { } second && NotEmpty(first) && NotEmpty(second))
```